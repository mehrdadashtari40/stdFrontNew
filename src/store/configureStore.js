import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';

import { userReducer, requestUserInfo } from '../common/user';
import { i18nReducer, i18nInit } from '../common/i18n';
import { trelloReducer } from '../views/trello';
import { systemManagementReducer } from '../views/system-management';
import { ProcessManagementReducer } from '../views/process-management';

import {reducer as layout2Reducer} from '../common/layout/_redux/layoutRedux'
import {reducer as tableReducer} from '../../src/common/tables/_redux/tableRedux'


import {reducer as basicInformationReducer} from '../views/basic-information/_redux/baseInfoRedux';
import LogsReducer from '../views/logs/LogReducesrs';

import BIReducer from '../views/BI/BIReducer';
import {reducer as inboxReducer} from "../views/inbox/_redux/inboxRedux";
import {reducer as authReducer} from '../../src/views/auth/_redux/authRedux'

import {reducer as ServiceDeskReducer} from "../views/service-desk/_redux/serviceDeskRedux";
import {reducer as SupervisorReducer} from "../views/supervisor/_redux/supervisorRedux";
import {reducer as CustomSettingsReducer} from "../views/customSettings/_redux/customSettingsRedux";
import {reducer as MenuReducer} from "../views/menu/_redux/menuRedux";
import {reducer as WorkDeskReducer} from "../views/work-desk/_redux/workDeskRedux";

export const rootReducer = combineReducers({
	auth: authReducer,
	user: userReducer,
	i18n: i18nReducer,
	basicInformation: basicInformationReducer,
	processManagement: ProcessManagementReducer,
	systemManagement: systemManagementReducer,
	ServiceDesk:ServiceDeskReducer,
	BI:BIReducer,
	trello:trelloReducer,
	inbox:inboxReducer,
	layout2:layout2Reducer,
	table:tableReducer,
	logs:LogsReducer,
	supervisor:SupervisorReducer,
	customSettingsReducer:CustomSettingsReducer,
	menuReducer:MenuReducer,
	workDeskReducer:WorkDeskReducer
});

const store = createStore(
	rootReducer,
	applyMiddleware(thunk)
);

store.dispatch(requestUserInfo());

store.dispatch(i18nInit());

export default store;
