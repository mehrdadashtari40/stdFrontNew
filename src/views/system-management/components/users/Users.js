import React from 'react';
import $ from 'jquery';
import {config} from '../../../../config/config';
import getAuthonticatedJSON from '../../../../common/utils/functions/getAuthenticatedJSON';
import putRequest from '../../../../common/utils/functions/putRequest';
import {smallBox, BigBreadcrumbs, WidgetGrid, JarvisWidget, SmartMessageBox} from '../../../../common';
import ServerSideDataTable from '../../../../common/tables/components/DataTableServerSide';
import Form from '../../utils/Form';
import jMoment from 'moment-jalaali';
import {connect} from 'react-redux';
import * as Actions from '../../SystemManagementActions';
import {withTranslation} from 'react-i18next';
import {compose} from 'redux';
import MaterialDataTable from "../../../../common/tables/components/MaterialDataTable";
import deleteAuthonticatedJSON from "../../../../common/utils/functions/deleteAuthonticatedJSON";
class Users extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            Users: [],
            refresh: false,
            total: null,
            limit: null,
            User: [],
            Edit: false,
            Show: [],
            filter: '',
            UserId: null,
            searchVal: '',
            status: '',
            successor: ''
        };
        this.handelCreate = this.handelCreate.bind(this);
        this.handelShow = this.handelShow.bind(this);
    }

    componentDidMount() {
        let self = this;
        $(document).ready(function () {
            // $(document).on('dblclick', 'td.row-controller.process-designer', OpenLink);
            $(document).on('dblclick', 'td.clicked-item', OpenLink);
            $('#modalClose').on('click', closeModal);

            function OpenLink() {
                let user_uid = $(this)
                    .siblings('td.clicked-item')
                    .find('input')
                    .attr('value');

                self.handelEdit(user_uid);

                self.setState({
                    Edit: true,
                });
            }

            function closeModal() {
                $('#UserBox').modal('hide');
                self.setNullFiels();
                self.setState({Edit: false, refresh: false});
            }
        });
    }

    getUsers = data => {
        this.setState({Users: data});
    };

    handelEdit(userUid) {
        $('#UserBox').modal('show');
        $('#actions').show();
        let self = this;
        this.props.loadUser(config.apiServer + 'user/' + userUid);

        getAuthonticatedJSON(config.apiServer + 'user/' + userUid, {
            headers: {
                Authorization: 'Bearer ' + localStorage.getItem('access_token'),
                'Content-Type': 'application/json',
            },
        }).then(res => {
            self.setState({Show: res, status: res.usr_status, successor: res.usr_replaced_by});
            $('#actions').show();
            this.setState({Show: res});
            $('#UserBox').modal('show');
            this.SetData();
        });
    }

    handelDelete(userUid) {
        SmartMessageBox(
            {
                title: this.props.t('Delete confirm title'),
                content: this.props.t('Delete confirm content message'),
                buttons: '[' + this.props.t('No') + '][' + this.props.t('Yes') + ']',
            },
            ButtonPressed => {
                if (ButtonPressed === this.props.t('Yes')) {
                    let self = this;
                    deleteAuthonticatedJSON(config.apiServer + 'user/' + userUid).then(res => {
                        if (res === 400) {
                            smallBox({
                                title: this.props.t('Warning'),
                                content:
                                    '<i>' + this.props.t('User can not deleted, user assign to the group') + '</i>',
                                color: '#cb8217',
                                iconSmall: 'fa fa-check fa-2x fadeInRight animated',
                                timeout: 4000,
                            });
                            document.getElementById('new-user-form').reset();
                            self.props.Refresh();
                            setTimeout(() => {
                                self.props.getRefreshStatus();
                            }, 500);
                        } else if (res === 200) {
                            smallBox({
                                title: this.props.t('Successfully'),
                                content: '<i>' + this.props.t('The user have been deleted successfully') + '</i>',
                                color: '#659265',
                                iconSmall: 'fa fa-check fa-2x fadeInRight animated',
                                timeout: 4000,
                            });
                            document.getElementById('new-user-form').reset();
                            let users = self.state.Users.filter(x=>x.USR_UID !== userUid );
                            self.getUsers(users)
                        }
                    })
                        .catch(error=>{
                            if (error.responseJSON.error !== undefined) {
                                var message = error.responseJSON.error.message;
                                if(message.substr(-46) === "cannot be deleted while it has cases assigned."){
                                    message = "User can not be deleted, user has open cases";
                                }
                                smallBox({
                                    title: this.props.t('Warning'),
                                    content:
                                        '<i>' + this.props.t(message) + '</i>',
                                    color: '#cb8217',
                                    iconSmall: 'fa fa-check fa-2x fadeInRight animated',
                                    timeout: 4000,
                                });
                            }
                        });
                }
            }
        );
    }

    handelIndex = () => {
        $('#UserBox').modal('hide');
        const refresh = this.state.refresh ? false : true;
        this.setState({
            refresh: refresh,
        });
    };

    getRefreshStatus = () => {
        this.setState({
            refresh: false,
        });
    };

    handelCreate() {
        this.setNullFiels();
        this.enableFields();
        $('#actions').hide();
        $('#UserBox').modal('show');
    }

    setNullFiels = () => {
        $('input[name="usr_username"]').val('');
        $('input[name="usr_firstname"]').val('');
        $('input[name="usr_lastname"]').val('');
        $('select[name="usr_role"]').val('');
        $('select[name="usr_status"]').val('');
        $('input[name="usr_email"]').val('');
        $('input[name="usr_phone"]').val('');
        $('input[name="usr_new_pass"]').val('');
        $('input[name="usr_cnf_pass"]').val('');
        $('input[name=usr_due_date]').val('');
    };

    handelShow(data) {
        getAuthonticatedJSON(config.apiServer + 'user/' + data['usr_uid'], {
            headers: {
                Authorization: 'Bearer ' + localStorage.getItem('access_token'),
                'Content-Type': 'application/json',
            },
        }).then(res => {
            $('#actions').show();
            this.setState({Show: res});
            $('#UserBox').modal('show');
            this.SetData();
            this.disableFields();
        });
    }

    SetData() {
        let data = this.props.User;
        $('input[name=usr_uid]').val(data['usr_uid']);
        $('input[name=usr_username]').val(data['usr_username']);
        $('input[name=usr_lastname]').val(data['usr_lastname']);
        $('input[name=usr_firstname]').val(data['usr_firstname']);
        $('input[name=usr_email]').val(data['usr_email']);
        $('input[name=usr_phone]').val(data['usr_phone']);
        $('select[name=usr_role]').val(data['usr_role']);
        $('select[name=usr_status]').val(data['usr_status']);
        $('select[name=usr_replaced_by]').val(data['usr_replaced_by']);
        $('input[name="usr_due_date"]').val(jMoment(data['usr_due_date']).format('jYYYY/jMM/jDD'));
        $('input[name=usr_new_pass]').val(data['usr_new_pass']);
        $('input[name=usr_cnf_pass]').val(data['usr_cnf_pass']);
    }

    disableFields() {
        $('input[name=usr_username]').prop('disabled', true);
        $('input[name=usr_lastname]').prop('disabled', true);
        $('input[name=usr_new_pass]').prop('disabled', true);
        $('input[name=usr_cnf_pass]').prop('disabled', true);
        $('input[name=usr_firstname]').prop('disabled', true);
        $('input[name=usr_email]').prop('disabled', true);
        $('input[name=usr_phone]').prop('disabled', true);
        $('select[name=usr_role]').prop('disabled', true);
        $('select[name=usr_status]').prop('disabled', true);
        $('input[name=usr_due_date]').prop('disabled', true);

        return false;
    }

    enableFields() {
        $('input[name=usr_username]').prop('disabled', false);
        $('input[name=usr_lastname]').prop('disabled', false);
        $('input[name=usr_firstname]').prop('disabled', false);
        $('input[name=usr_email]').prop('disabled', false);
        $('input[name=usr_new_pass]').prop('disabled', false);
        $('input[name=usr_cnf_pass]').prop('disabled', false);
        $('input[name=usr_phone]').prop('disabled', false);
        $('select[name=usr_role]').prop('disabled', false);
        $('select[name=usr_status]').prop('disabled', false);
        $('input[name=usr_due_date]').prop('disabled', false);

        return false;
    }

    onSubmit = evt => {
        evt.preventDefault();
        if (!this.checkPasswordLength()) {
            return;
        }

        if (!this.checkConfirmPassword()) {
            return;
        }
        const data = {
            usr_new_pass: $('#ch_usr_new_pass').val(),
        };
        putRequest(config.apiServer + `users/${this.state.UserId}`, {
            body: JSON.stringify(data),
            headers: {
                Authorization: 'Bearer ' + localStorage.getItem('access_token'),
                'Content-Type': 'application/json',
            },
        }).then(res => {
            smallBox({
                title: this.props.t('Successfully'),
                content: '<i>' + this.props.t('Data has been saved successfully') + '</i>',
                color: '#659265',
                iconSmall: 'fa fa-check fa-2x fadeInRight animated',
                timeout: 4000,
            });
            document.getElementById('change-user-password').reset();
        });
    };

    searchUser = e => {
        this.setState({
            searchVal: e.target.value,
        });
    };

    render() {
        let users = null;
        if (this.state.Users.length > 0) {
            var tmp = [];
            this.state.Users.map(user => {
                tmp.push(this.getTableRow(user,this));
            });
            users = tmp;
        }
        console.log(process.env.REACT_APP_NOT_SECRET_CODE)

        return (<><MaterialDataTable
            title={this.props.t('Users')}
            icon={'fas fa-users'}
            url={config.apiServer + 'users/anr-list'}
            columns={this.getTableRow(users, this)}
            main_filter={true}
            getDataList={data => this.getUsers(data)}
            toolbar_btns={<>
                <button className="btn btn_inbox"
                        onClick={this.handelCreate}
                >
                    <i className="fa fa-plus" />
                    <span>{this.props.t('New user')}</span>
                </button>
            </>}
        />
            <div
                className="modal fade-in-up"
                id="UserBox"
                role="dialog"
                data-keyboard="false"
                data-backdrop="static"
            >
                <div className="modal-dialog modal-lg">
                    <div className="modal-content" id="UserBoxBody">
                        {/*{editUser}*/}
                        {/*<Form Refresh={this.handelIndex}*/}
                        <Form
                            AddUsers={(data)=>{
                                let users = this.state.Users;

                                var key, keys = Object.keys(data);
                                var n = keys.length;
                                var newUser={}
                                while (n--) {
                                    key = keys[n];
                                    newUser[key.toLowerCase()] = data[key];
                                }
                                users.push(newUser);
                                this.getUsers(users)
                            }}
                            removeUser={(uid)=>{
                                let users = this.state.Users.filter(x=>x.USR_UID !== uid );
                                this.getUsers(users)
                            }}

                            Refresh={this.handelIndex}
                            getRefreshStatus={this.getRefreshStatus}
                            Edit={this.state.Edit}
                            Disable={this.disableFields}
                            Enable={this.enableFields}
                            Data={this.props.User}
                            status={this.state.status}
                            successor={this.state.successor}
                        />
                    </div>
                </div>
            </div>
            <div
                className="modal fade-in-up"
                id="changePassword"
                role="dialog"
                data-keyboard="false"
                data-backdrop="static"
            >
                <div className="modal-dialog modal-lg">
                    <div className="modal-content">
                        <form
                            id="change-user-password"
                            className="smart-form"
                            noValidate="novalidate"
                            onSubmit={this.onSubmit}
                        >
                            <div className="modal-header">
                                {/*<button type="button" className="close" data-dismiss="modal">&times;</button>*/}
                                <button
                                    type="button"
                                    className="close"
                                    data-dismiss="modal"
                                    aria-hidden="true"
                                    id="modalClose"
                                >
                                    &times;
                                </button>
                                <h4 className="modal-title">{this.props.t('Change Password')}</h4>
                            </div>
                            <div>
                                <fieldset>
                                    <div className="row">
                                        <section className="col col-6">
                                            <label>
                                                {this.props.t('Password')}
                                                <strong style={{color: 'red'}}>*</strong>
                                            </label>
                                            <label className="input">
                                                <i className="icon-prepend fa fa-lock"/>
                                                <input
                                                    type="password"
                                                    name="usr_new_pass"
                                                    id="ch_usr_new_pass"
                                                    placeholder={this.props.t('Password')}
                                                />
                                            </label>
                                            <div
                                                className="help-block text-danger"
                                                style={{color: '#d44950'}}
                                                id="passwordValidationChangeMsg"
                                            ></div>
                                        </section>
                                        <section className="col col-6">
                                            <label>
                                                {this.props.t('Password Confirmation')}
                                                <strong style={{color: 'red'}}>*</strong>
                                            </label>
                                            <label className="input">
                                                <i className="icon-prepend fa fa-lock"/>
                                                <input
                                                    type="password"
                                                    name="usr_cnf_pass"
                                                    id="ch_usr_cnf_pass"
                                                    placeholder={this.props.t('Password Confirmation')}
                                                />
                                            </label>
                                            <div
                                                className="help-block text-danger"
                                                style={{color: '#d44950'}}
                                                id="confirmPasswordChangeValidationMsg"
                                            ></div>
                                        </section>
                                    </div>
                                </fieldset>
                            </div>
                            <footer>
                                <button type="submit" className="btn btn-success">
                                    <i className="fa fa-check"></i>
                                    {this.props.t('Submit')}
                                </button>
                            </footer>
                        </form>
                    </div>
                </div>
            </div>
        </>);

    }

    getTableRow = (item, self) => {
        return [
            {
                dataField: 'usr_username',
                text: self.props.t('User Name'),
            },
            {
                dataField: 'usr_name',
                text: self.props.t('Last Name'),
                formatter: function (cell, row) {
                    return (<>
                        {row.usr_firstname} {row.usr_lastname}
                    </>)
                }
            },
            {
                dataField: 'usr_status',
                text: self.props.t('Status'),
                formatter: function (cell, item) {
                    return (<>
                        <span className="label label-success">{self.props.t(item.usr_status.toUpperCase())}</span>

                    </>)
                }
            },
            {
                dataField: 'usr_replaced_by',
                text: self.props.t('Supervisor'),
                formatter: function (cell, item) {
                    if (item.usr_replaced_by === "")
                        return self.props.t('Not selected');
                    return item.usr_replaced_by
                }
            },
            {
                dataField: 'role',
                text: self.props.t('Role'),
                formatter: function (cell, item) {
                    let role = '';
                    if (item.usr_role === 'PROCESSMAKER_ADMIN') role = self.props.t('Admin');
                    else if (item.usr_role === 'PROCESSMAKER_OPERATOR') role = self.props.t('Operator');
                    else if (item.usr_role === 'PROCESSMAKER_MANAGER') role = self.props.t('Manager');
                    else if (item.usr_role === 'PROCESSMAKER_DEPADMIN') role = self.props.t('DepAdmin');
                    return (<>
                        {role}
                    </>)
                }
            },
            {
                dataField: 'departmentname',
                text: self.props.t('Department'),
            },
            {
                dataField: 'last_login',
                text: self.props.t('Last Login'),
                formatter: function (cell, item) {
                    let date = jMoment(item.usr_update_date).format('jYYYY/jMM/jDD');
                    let time = jMoment(item.usr_update_date).format('HH:mm:ss');

                    return (<>
                        {time + ' ' + date}
                    </>)
                }
            },
            {
                dataField: 'actions',
                text: self.props.t('actions'),
                formatter: function (cell, item) {
                    return (<div className={'table-actions'}>
                            <i className={'fas fa-edit'}
                                onClick={()=>{
                                    self.handelEdit(item.usr_uid);
                                    self.setState({
                                        Edit: true,
                                    });
                                }}
                            />
                            <i className={'fas fa-key'}
                               onClick={() => self.editUser(item.usr_uid)}
                            />
                            <i className={'fas fa-trash'}
                               onClick={() => self.handelDelete(item.usr_uid)}
                            />
                    </div>)
                }
            },
        ]
    };

    editUser = userUID => {
        $('#changePassword').modal('show');
        this.setState({UserId: userUID});
        //this.checkConfirmPassword();
    };

    checkConfirmPassword = () => {
        // $(passwordValidationMsg)
        const password = $('#ch_usr_new_pass').val();
        const confirmPassword = $('#ch_usr_cnf_pass').val();
        if (password == confirmPassword) {
            return true;
        } else {
            $('#confirmPasswordChangeValidationMsg').text(this.props.t('Confirm Password'));
            setTimeout(() => {
                $('#confirmPasswordChangeValidationMsg').text('');
            }, 20000);
            return false;
        }
    };

    checkPasswordLength = () => {
        // $(passwordValidationMsg)
        // let password = $('#ch_usr_new_pass').val();
        let password = $('#ch_usr_new_pass').val();

        if (password.length >= 6) {
            // console.log('ch : length tr :: ', password);
            return true;
        } else {
            // console.log('ch : length fa :: ', password);
            $('#passwordValidationChangeMsg').text(this.props.t('Minlength Password'));
            setTimeout(() => {
                $('#passwordValidationChangeMsg').text('');
            }, 20000);
            return false;
        }
    };

    filterUsers = e => {
        // this.get
        let filter = $('#searchBox').val();
        this.setState({filter: filter, refresh: true});
    };
}

export default compose(
    withTranslation(),
    connect(state => state.systemManagement, Actions)
)(Users);
