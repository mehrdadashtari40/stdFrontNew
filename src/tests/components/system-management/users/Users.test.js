import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../../store/configureStore';
import configureStore from 'redux-mock-store';
import * as selectActions from '../../../../views/system-management/SystemManagementActions';
import Users from '../../../../views/system-management/components/users/Users';

// const store = mockStore();

const mockUserData = {
	dep_uid: '8de55952f1d8a7c5f6ed226671bd6c39',
	departmentname: '24.بیمارستان الغدیر ابهر',
	usr_address: 'علوم پزشکی زنجان',
	usr_birthday: null,
	usr_calendar_name: '',
	usr_calendar_uid: '',
	usr_cellular: '',
	usr_city: 'Zn',
	usr_country: 'IR',
	usr_create_date: '2019-04-04 00:00:00',
	usr_due_date: '2099-01-01',
	usr_email: '0061954349',
	usr_fax: '--',
	usr_firstname: '0061954349',
	usr_lastname: '0061954349',
	usr_location: '',
	usr_phone: "'0098'",
	usr_photo_path: '/var/www/html/processmaker/workflow/public_html/images/user.gif',
	usr_position: '',
	usr_replaced_by: '',
	usr_reports_to: '0061954349',
	usr_resume: '0061954349',
	usr_role: 'PROCESSMAKER_ADMIN',
	usr_status: 'ACTIVE',
	usr_uid: '0061954349',
	usr_update_date: '2019-04-04 00:00:00',
	usr_username: '0061954349',
	usr_ux: 'NORMAL',
	usr_zip_code: "'1'",
};

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));

test('Make Snapshot file , Users Component', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(wrapper).toMatchSnapshot();
});

it('invokes `componentDidMount` when mounted', () => {
	jest.spyOn(Users.prototype, 'componentDidMount');
	shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(Users.prototype.componentDidMount).toHaveBeenCalled();
	Users.prototype.componentDidMount.mockRestore();
});


it('searchUser HaveBeenCalled when searchBox change ?', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	const spy = jest.spyOn(instance, 'searchUser');

	instance.forceUpdate();

	const input = wrapper.find('#searchBox');
	input.simulate('change', { target: { value: 'test' } });
	expect(spy).toHaveBeenCalled();
});

it('filterUsers HaveBeenCalled ?', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	const spy = jest.spyOn(instance, 'filterUsers');

	instance.forceUpdate();

	const Button = wrapper.find('#filterUser');
	Button.simulate('click');
	expect(spy).toHaveBeenCalled();
});


test('getUsers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.getUsers).toBeDefined();
});

test('handelEdit Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelEdit).toBeDefined();
});

test('handelIndex Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelIndex).toBeDefined();
});

test('getRefreshStatus Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.getRefreshStatus).toBeDefined();
});

test('handelCreate Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelCreate).toBeDefined();
});

test('setNullFiels Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.setNullFiels).toBeDefined();
});

test('handelShow Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handelShow).toBeDefined();
});

test('SetData Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.SetData).toBeDefined();
});

test('disableFields Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.disableFields).toBeDefined();
});

test('enableFields Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.enableFields).toBeDefined();
});

test('onSubmit Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Users />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.onSubmit).toBeDefined();
});

describe('actions', () => {
	it('should create an action to loadUser', () => {
		const data = mockUserData;
		const expectedAction = {
			type: selectActions.LOAD_USER,
			User: data,
		};
		expect(selectActions.setUser(data)).toEqual(expectedAction);
	});
});
