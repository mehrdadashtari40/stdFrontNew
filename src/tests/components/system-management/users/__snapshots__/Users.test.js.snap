// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , Users Component 1`] = `
<div
  id="content"
>
  <div
    className="row"
  >
    <BigBreadcrumbs
      className="col-xs-12 col-sm-7 col-md-7 col-lg-4"
      icon="fa fa-fw fa-users"
      items={
        Array [
          "کاربران",
          "مدیریت کاربران",
        ]
      }
    />
  </div>
  <WidgetGrid>
    <div
      className="row"
    >
      <article
        className="col-sm-12"
      >
        <JarvisWidget
          collapsed={false}
          color={false}
          colorbutton={false}
          custombutton={false}
          deletebutton={false}
          editbutton={false}
          fullscreenbutton={true}
          hidden={false}
          load={false}
          refresh={false}
          sortable={true}
          togglebutton={false}
        >
          <header>
            <span
              className="widget-icon"
            >
               
              <i
                className="fa fa-bars"
              />
               
            </span>
            <h2>
               
            </h2>
            <div
              className="widget-toolbar"
            >
              <a
                className="btn btn-outline-success little_btn"
                onClick={[Function]}
              >
                <i
                  className="fa fa-plus"
                />
                <span>
                   
                </span>
              </a>
            </div>
          </header>
          <div
            className="tab-content"
            id="myTabContent1"
            role="content"
          >
            <div
              className="tab-pane fade in active"
            >
              <div
                className="dataTables_wrapper form-inline dt-bootstrap no-footer"
                id="DataTables_Table_7_wrapper"
              >
                <DataTableServerSide
                  filter=""
                  getDataList={[Function]}
                  refresh={false}
                  url="http://ls.arian.co.ir:8081/api/1.0/zums/users/anr-list"
                >
                  <thead>
                    <tr
                      role="row"
                    >
                      <th
                        colSpan="7"
                      >
                        <div
                          className="row"
                        >
                          <div
                            className="col col-md-11"
                          >
                            <input
                              className="form-control searchBox"
                              id="searchBox"
                              onChange={[Function]}
                              placeholder=""
                              style={
                                Object {
                                  "width": "100%",
                                }
                              }
                              type="text"
                            />
                          </div>
                          <div
                            className="col col-md-1"
                          >
                            <button
                              className="btn btn-success"
                              id="filterUser"
                              onClick={[Function]}
                            >
                              جستجو
                            </button>
                          </div>
                        </div>
                      </th>
                    </tr>
                    <tr
                      role="row"
                    >
                      <th
                        aria-controls="DataTables_Table_7"
                        aria-label="نام کاربری: مرتب سازی"
                        aria-sort="ascending"
                        className="sorting_asc"
                        colspan="1"
                        data-hide="phone"
                        rowspan="1"
                        tabindex="0"
                      >
                        <i
                          className="fa fa-fw fa-user text-muted hidden-md hidden-sm hidden-xs"
                        />
                        نام کاربری
                      </th>
                      <th
                        aria-controls="DataTables_Table_7"
                        aria-label="نام و نام خانوادگی: مرتب سازی"
                        className="expand sorting"
                        colspan="1"
                        data-className="expand"
                        rowspan="1"
                        tabindex="0"
                      >
                        <i
                          className="fa fa-fw fa-user text-muted hidden-md hidden-sm hidden-xs"
                        />
                        نام و نام خانوادگی
                      </th>
                      <th
                        abindex="0"
                        aria-controls="DataTables_Table_7"
                        aria-label="وضعیت: مرتب سازی"
                        className="sorting"
                        colspan="1"
                        data-hide="phone"
                        rowspan="1"
                      >
                        <i
                          className="fa fa-fw fa-star text-muted hidden-md hidden-sm hidden-xs"
                        />
                        وضعیت
                      </th>
                      <th
                        aria-controls="DataTables_Table_7"
                        aria-label="نقش: مرتب سازی"
                        className="sorting"
                        colspan="1"
                        rowspan="1"
                        tabindex="0"
                      >
                        نقش
                      </th>
                      <th
                        aria-controls="DataTables_Table_7"
                        aria-label="دپارتمان: مرتب سازی"
                        className="sorting"
                        colSpan="1"
                        rowSpan="1"
                        tabIndex="0"
                      >
                        دپارتمان
                      </th>
                      <th
                        aria-controls="DataTables_Table_7"
                        aria-label="آخرین ورود: مرتب سازی"
                        className="sorting"
                        colspan="1"
                        data-hide="phone,tablet"
                        rowspan="1"
                        tabindex="0"
                      >
                        <i
                          className="fa fa-fw fa-rotate-left txt-color-blue hidden-md hidden-sm hidden-xs"
                        />
                        آخرین ورود
                      </th>
                      <th
                        aria-controls="DataTables_Table_7"
                        aria-label="آخرین ورود: مرتب سازی"
                        className="sorting"
                        colspan="1"
                        data-hide="phone,tablet"
                        rowspan="1"
                        tabindex="0"
                      >
                        <i
                          className="fa fa-cog fa-rotate-left txt-color-blue hidden-md hidden-sm hidden-xs"
                        />
                        عملیات
                      </th>
                    </tr>
                  </thead>
                  <tbody />
                </DataTableServerSide>
              </div>
            </div>
          </div>
        </JarvisWidget>
      </article>
    </div>
  </WidgetGrid>
  <div
    className="modal fade-in-up"
    data-backdrop="static"
    data-keyboard="false"
    id="UserBox"
    role="dialog"
  >
    <div
      className="modal-dialog modal-lg"
    >
      <div
        className="modal-content"
        id="UserBoxBody"
      >
        <Form
          Data={Array []}
          Disable={[Function]}
          Edit={false}
          Enable={[Function]}
          Refresh={[Function]}
          getRefreshStatus={[Function]}
          t={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="modal fade-in-up"
    data-backdrop="static"
    data-keyboard="false"
    id="changePassword"
    role="dialog"
  >
    <div
      className="modal-dialog modal-lg"
    >
      <div
        className="modal-content"
      >
        <form
          className="smart-form"
          id="change-user-password"
          noValidate="novalidate"
          onSubmit={[Function]}
        >
          <div
            className="modal-header"
          >
            <button
              aria-hidden="true"
              className="close"
              data-dismiss="modal"
              id="modalClose"
              type="button"
            >
              ×
            </button>
            <h4
              className="modal-title"
            />
          </div>
          <div>
            <fieldset>
              <div
                className="row"
              >
                <section
                  className="col col-6"
                >
                  <label>
                    <strong
                      style={
                        Object {
                          "color": "red",
                        }
                      }
                    >
                      *
                    </strong>
                  </label>
                  <label
                    className="input"
                  >
                    <i
                      className="icon-prepend fa fa-lock"
                    />
                    <input
                      id="ch_usr_new_pass"
                      name="usr_new_pass"
                      placeholder=""
                      type="password"
                    />
                  </label>
                  <div
                    className="help-block text-danger"
                    id="passwordValidationChangeMsg"
                    style={
                      Object {
                        "color": "#d44950",
                      }
                    }
                  />
                </section>
                <section
                  className="col col-6"
                >
                  <label>
                    <strong
                      style={
                        Object {
                          "color": "red",
                        }
                      }
                    >
                      *
                    </strong>
                  </label>
                  <label
                    className="input"
                  >
                    <i
                      className="icon-prepend fa fa-lock"
                    />
                    <input
                      id="ch_usr_cnf_pass"
                      name="usr_cnf_pass"
                      placeholder=""
                      type="password"
                    />
                  </label>
                  <div
                    className="help-block text-danger"
                    id="confirmPasswordChangeValidationMsg"
                    style={
                      Object {
                        "color": "#d44950",
                      }
                    }
                  />
                </section>
              </div>
            </fieldset>
          </div>
          <footer>
            <button
              className="btn btn-success"
              type="submit"
            >
              <i
                className="fa fa-check"
              />
            </button>
          </footer>
        </form>
      </div>
    </div>
  </div>
</div>
`;
